File: springboot2.x-learning/springboot-data-jpa/src/main/java/com/rtxtitanv/model/query/UserQuery.java
package com.rtxtitanv.model.query;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

import java.util.List;

/**
 * @author rtxtitanv
 * @version 1.0.0
 * @name com.rtxtitanv.model.query.UserQuery
 * @description 封装用户查询条件类
 * @date 2020/1/3 15:23
 */
@Accessors(chain = true)
@AllArgsConstructor
@NoArgsConstructor
@Data
public class UserQuery {

    private Long idQuery;
    private String userNameQuery;
    private String nickNameQuery;
    private String passWordQuery;
    private String emailQuery;
    private String telQuery;
    private Integer minAgeQuery;
    private Integer maxAgeQuery;
    private List<Long> idsQuery;
    private List<Integer> agesQuery;

}

File: springboot2.x-learning/springboot-data-jpa-kotlin/src/main/kotlin/com/rtxtitanv/model/query/UserQuery.kt
package com.rtxtitanv.model.query

/**
 * @name com.rtxtitanv.model.query.UserQuery
 * @description 封装用户查询条件类
 * @author rtxtitanv
 * @date 2020/1/17 23:55
 * @version v1.0.0
 */
data class UserQuery(var idQuery: Long? = null,
                     var userNameQuery: String? = null,
                     var nickNameQuery: String? = null,
                     var passWordQuery: String? = null,
                     var emailQuery: String? = null,
                     var telQuery: String? = null,
                     var minAgeQuery: Int? = null,
                     var maxAgeQuery: Int? = null,
                     var idsQuery: List<Long>? = null,
                     var agesQuery: List<Int>? = null)